src/sinks/aws_sqs.rs

```diff
@@ -250,6 +265,7 @@ fn encode_event(
     mut event: Event,
     encoding: &EncodingConfig<Encoding>,
     message_group_id: Option<&Template>,
+    message_deduplication_id: Option<&Template>,
```

Comment:
Could these be type `&Option<Template>` instead? That avoids the need for `as_ref` above and shouldn't complicate the match below significantly.

Reply:
👍 is this common practice for `Option`s? It seems a bit weird when passing values in directly to have to do `&None`.

I made the change in ac88704fb